tests/engine/adhoc_test.go

```diff
@@ -221,3 +225,41 @@ func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
 	require.NoError(t, err)
 	require.Equal(t, uint64(42), results[0])
 }
+
+// Test that host functions can handle float parameter without corrupting the value
+func hostFuncWithFloat32Param(t *testing.T, newEngine func() wasm.Engine) {
+	ctx := context.Background()
+	mod, err := text.DecodeModule([]byte(`(module
+		;; 'test_f32param' is a host function which accepts a f32 param
```

Comment:
since import implies it is a host function (in a single module setup anyway) and the signature says itself anyway we can skip clarifying these things. focused comments will end up winning. Not sure if it is normal or not to quote the function name, so feel free to adjust:

`;; test_f32param crashes the runtime if the parameter is unexpected.

Reply:
`test_f32param` is replaced with `identity_float` now, and I skipped commenting about its signature as suggested.